xivo/tests/test_pubsub.py

```diff
@@ -120,3 +121,65 @@ def test_when_exception_then_other_callbacks_are_run(self):
 
         assert_that(callback_1.called)
         assert_that(callback_3.called)
+
+
+class TestCallbackCollector(unittest.TestCase):
```

Comment:
Could you add a few more tests?
- collecting unknown source = does nothing
- publish can handle exception of a subscriber

Reply:
I set `publish` and `collect`as private method since there are only used internally.
With this modification, there are no use case to collect with an unknown source id